kernel/src/platform/platform.rs

```diff
@@ -0,0 +1,171 @@
+//! Interfaces for implementing boards in Tock.
+
+use crate::errorcode;
+use crate::process;
+use crate::scheduler::Scheduler;
+use crate::syscall;
+use crate::syscall_driver::SyscallDriver;
+
+use crate::platform::chip::Chip;
+use crate::platform::scheduler_timer;
+use crate::platform::watchdog;
+
+/// Combination trait that boards provide to the kernel that includes all of
+/// the extensible operations the kernel supports.
+///
+/// This is the primary method for configuring the kernel for a specific board.
+pub trait KernelResources<C: Chip> {
+    /// The implementation of the system call dispatch mechanism the kernel
+    /// will use.
+    type SystemCallDispatch: SystemCallDispatch;
```

Comment:
Looking at this at first made me think this should type should provide the logic of actually dispatching system call. Maybe renaming it to something like `SystemCallDriverRegister` might offer a better suggestion of what this data type is supposed to do.

Reply:
I agree this might be misleading. In practice this will route most system calls but not all, `memop` and `yield` are important exceptions.

Furthermore, can we replace `Dispatch` with `Router`, as we're really only determining the `Driver` to use here?

Reply:
I'm not opposed to this change, and it should be quite easy to make at any point. But we discussed this https://github.com/tock/tock/pull/2551 and dispatch was the name, so I'm waiting if other's have a preference since this will end up in every board main.rs.

Reply:
Coming from the networking perspective, "routing" is not a good choice. How about "SystemCallLookup"? Or "SyscallDriverLookup"?

Reply:
My vote is for `SyscallDriverLookup`